lib/determinator/control.rb

```diff
@@ -115,12 +116,37 @@ def choose_target_group(feature, properties)
 
         tg.constraints.reduce(true) do |fit, (scope, *required)|
           present = [*normalised_properties[scope]]
-          fit && (required.flatten & present.flatten).any?
+          fit && (has_any(required, present) || has_any_app_version?(scope, required, present))
         end
-      # Must choose target group deterministically, if more than one match
+        # Must choose target group deterministically, if more than one match
       }.sort_by { |tg| tg.rollout }.last
     end
 
+    def has_any(required, present)
+      (required.flatten & present.flatten).any?
+    end
+
+    def has_any_app_version?(scope, required, present)
+      return false unless scope == 'app_version'
+
+      present.flatten.select {|version|
+        begin
+          present_sem_version = Semantic::Version.new version
+        rescue
+          # Ignore malformed app_version value
+          next false
```

Comment:
I'm not sure what we usually do in those scenario? Ignore or raise an exception?

Reply:
A malconfigured flag shouldn't raise, so just ignore.

Reply:
I reckon we actually might _want_ to raise here; Determinator [catches errors, returning false](https://github.com/deliveroo/determinator/blob/master/lib/determinator/control.rb#L99-L105) and tracking the error using whatever system the implementor specifies, this means the fact that the app version or feature details are invalid will be tracked, but will make sure that the given flag will always return `false`, which I think is a good outcome for something unexpected.

Reply:
Ah, in the file JP links:

```
rescue ArgumentError
      raise
```

So, I think you'll want to catch the `ArgumentError` here and re-raise as a `NotASemvar` error (or something similar)